File: tests/fixtures-service/http_clients.go
package svc

import "net/http"

// http.Get — should record GET /api/users via framework=http.
func ListUsers() (*http.Response, error) {
	return http.Get("/api/users")
}

// http.Post — POST /api/users via framework=http.
func CreateUser(body interface{}) (*http.Response, error) {
	return http.Post("/api/users", "application/json", nil)
}

// client.Get — generic http-client framework.
func FetchOrders(client *http.Client) (*http.Response, error) {
	return client.Get("/api/orders")
}

// http.NewRequest("POST", ...) — method derived from first string arg.
func MakeRequest() (*http.Request, error) {
	return http.NewRequest("POST", "/api/items", nil)
}

File: tests/fixtures-service/HttpClients.java
package svc;

import java.net.URI;
import java.net.http.HttpRequest;
import org.springframework.web.client.RestTemplate;

public class HttpClients {
    private final RestTemplate restTemplate = new RestTemplate();

    // RestTemplate getForObject — should record GET /api/users via spring-rest.
    public String listUsers() {
        return restTemplate.getForObject("/api/users", String.class);
    }

    // postForObject — POST /api/orders via spring-rest.
    public String createOrder(Object body) {
        return restTemplate.postForObject("/api/orders", body, String.class);
    }

    // java.net.http HttpRequest.newBuilder(URI.create("..."))
    public HttpRequest buildPing() {
        return HttpRequest.newBuilder(URI.create("https://payment-service/api/ping")).build();
    }
}
